[2025-09-16T17:52:05.038+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: opensky_data_pipeline_dag.run_kafka_consumer_slow scheduled__2025-09-16T17:45:00+00:00 [queued]>
[2025-09-16T17:52:05.041+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: opensky_data_pipeline_dag.run_kafka_consumer_slow scheduled__2025-09-16T17:45:00+00:00 [queued]>
[2025-09-16T17:52:05.041+0000] {taskinstance.py:2170} INFO - Starting attempt 1 of 1
[2025-09-16T17:52:05.049+0000] {taskinstance.py:2191} INFO - Executing <Task(PythonOperator): run_kafka_consumer_slow> on 2025-09-16 17:45:00+00:00
[2025-09-16T17:52:05.052+0000] {standard_task_runner.py:60} INFO - Started process 10069 to run task
[2025-09-16T17:52:05.053+0000] {standard_task_runner.py:87} INFO - Running: ['airflow', 'tasks', 'run', 'opensky_data_pipeline_dag', 'run_kafka_consumer_slow', 'scheduled__2025-09-16T17:45:00+00:00', '--job-id', '138', '--raw', '--subdir', 'DAGS_FOLDER/opensky_data_pipeline_dag.py', '--cfg-path', '/tmp/tmpif9112i4']
[2025-09-16T17:52:05.054+0000] {standard_task_runner.py:88} INFO - Job 138: Subtask run_kafka_consumer_slow
[2025-09-16T17:52:05.076+0000] {task_command.py:423} INFO - Running <TaskInstance: opensky_data_pipeline_dag.run_kafka_consumer_slow scheduled__2025-09-16T17:45:00+00:00 [running]> on host 64220da1adfe
[2025-09-16T17:52:05.184+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='opensky_data_pipeline_dag' AIRFLOW_CTX_TASK_ID='run_kafka_consumer_slow' AIRFLOW_CTX_EXECUTION_DATE='2025-09-16T17:45:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-09-16T17:45:00+00:00'
[2025-09-16T17:52:05.185+0000] {kafka_consumer.py:37} INFO - Starting receiving process...
[2025-09-16T17:52:05.186+0000] {conn.py:396} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=kafka:9092 <connecting> [IPv4 ('172.18.0.4', 9092)]>: connecting to kafka:9092 [('172.18.0.4', 9092) IPv4]
[2025-09-16T17:52:05.186+0000] {conn.py:618} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=kafka:9092 <checking_api_versions_recv> [IPv4 ('172.18.0.4', 9092)]>: Broker version identified as 2.6
[2025-09-16T17:52:05.186+0000] {conn.py:457} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=kafka:9092 <connected> [IPv4 ('172.18.0.4', 9092)]>: Connection complete.
[2025-09-16T17:52:05.187+0000] {subscription_state.py:181} INFO - Updating subscribed topics to: ('opensky_data',)
[2025-09-16T17:52:05.187+0000] {conn.py:396} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=kafka:9092 <connecting> [IPv4 ('172.18.0.4', 9092)]>: connecting to kafka:9092 [('172.18.0.4', 9092) IPv4]
[2025-09-16T17:52:05.188+0000] {conn.py:457} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=1 host=kafka:9092 <connected> [IPv4 ('172.18.0.4', 9092)]>: Connection complete.
[2025-09-16T17:52:05.188+0000] {conn.py:945} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=bootstrap-0 host=kafka:9092 <connected> [IPv4 ('172.18.0.4', 9092)]>: Closing connection. 
[2025-09-16T17:52:05.290+0000] {cluster.py:401} INFO - Coordinator for group/opensky_full_pipeline_consumer_01 is ('coordinator-1', 'kafka', 9092, None)
[2025-09-16T17:52:05.291+0000] {base.py:788} INFO - Discovered coordinator coordinator-1 for group opensky_full_pipeline_consumer_01
[2025-09-16T17:52:05.291+0000] {base.py:848} INFO - Starting new heartbeat thread
[2025-09-16T17:52:05.291+0000] {consumer.py:365} INFO - Revoking previously assigned partitions () for group opensky_full_pipeline_consumer_01
[2025-09-16T17:52:05.291+0000] {base.py:419} INFO - Failed to join group opensky_full_pipeline_consumer_01: NodeNotReadyError: coordinator-1
[2025-09-16T17:52:05.292+0000] {conn.py:396} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=coordinator-1 host=kafka:9092 <connecting> [IPv4 ('172.18.0.4', 9092)]>: connecting to kafka:9092 [('172.18.0.4', 9092) IPv4]
[2025-09-16T17:52:05.292+0000] {conn.py:457} INFO - <BrokerConnection client_id=kafka-python-2.2.15, node_id=coordinator-1 host=kafka:9092 <connected> [IPv4 ('172.18.0.4', 9092)]>: Connection complete.
[2025-09-16T17:52:05.397+0000] {base.py:532} INFO - (Re-)joining group opensky_full_pipeline_consumer_01
[2025-09-16T17:52:05.398+0000] {base.py:637} INFO - Received member id kafka-python-2.2.15-ebf48412-5b4a-47a7-bf65-89a209ee8110 for group opensky_full_pipeline_consumer_01; will retry join-group
[2025-09-16T17:52:05.398+0000] {base.py:419} INFO - Failed to join group opensky_full_pipeline_consumer_01: [Error 79] MemberIdRequiredError
[2025-09-16T17:52:05.398+0000] {base.py:532} INFO - (Re-)joining group opensky_full_pipeline_consumer_01
[2025-09-16T17:52:08.366+0000] {base.py:594} INFO - Successfully joined group opensky_full_pipeline_consumer_01 <Generation 94 (member_id: kafka-python-2.2.15-ebf48412-5b4a-47a7-bf65-89a209ee8110, protocol: range)>
[2025-09-16T17:52:08.372+0000] {subscription_state.py:254} INFO - Updated partition assignment: [('opensky_data', 0)]
[2025-09-16T17:52:08.372+0000] {consumer.py:249} INFO - Setting newly assigned partitions (('opensky_data', 0),) for group opensky_full_pipeline_consumer_01
[2025-09-16T17:52:19.267+0000] {kafka_consumer.py:77} INFO - Saved 69876 rows to Parquet at /opt/airflow/scripts/opensky/../../data/kafka_parquet/opensky_batch_20250916_175219.parquet
[2025-09-16T17:52:19.267+0000] {base.py:1052} INFO - Stopping heartbeat thread
[2025-09-16T17:52:19.276+0000] {python.py:201} INFO - Done. Returned value was: /opt/airflow/scripts/opensky/../../data/kafka_parquet/opensky_batch_20250916_175219.parquet
[2025-09-16T17:52:19.286+0000] {taskinstance.py:1138} INFO - Marking task as SUCCESS. dag_id=opensky_data_pipeline_dag, task_id=run_kafka_consumer_slow, execution_date=20250916T174500, start_date=20250916T175205, end_date=20250916T175219
[2025-09-16T17:52:19.312+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 0
[2025-09-16T17:52:19.395+0000] {taskinstance.py:3280} INFO - 1 downstream tasks scheduled from follow-on schedule check
